py_gtfs_rt_ingestion/batch_files.py

```diff
@@ -45,35 +47,44 @@ def parseArgs(args) -> dict:
         help='filesize threshold for batch sizes')
 
     parser.add_argument(
-        '--pretty',
-        dest='pretty',
+        '--dry-run',
+        dest='dry_run',
         action='store_true',
-        help='print out a pretty summary of the batches')
+        help='do not invoke ingest lambda function')
+
+    parser.add_argument(
+        '--print-events',
+        dest='print_events',
+        action='store_true',
+        help='print out each event as json to stdout')
 
     return vars(parser.parse_args(args))
 
 def main(batch_args: BatchArgs) -> None:
     file_list = file_list_from_s3(bucket_name=batch_args.s3_bucket,
                                   file_prefix=batch_args.s3_prefix)
 
-    batches = batch_files(file_list, batch_args.filesize_threshold)
 
-    # TODO use boto3 to launch ingestion jobs from each batch
+    total_bytes = 0
+    total_files = 0
+    events = []
+    for batch in batch_files(file_list, batch_args.filesize_threshold):
+        total_bytes += batch.total_size
+        total_files += len(batch.filenames)
+        events.append(batch.create_event())
 
-    # log out a summary of whats been batched.
-    if batch_args.pretty:
-        total_bytes = 0
-        total_files = 0
-        for batch in batches:
-            total_bytes += batch.total_size
-            total_files += len(batch.filenames)
+        if not batch_args.dry_run:
+            try:
+                batch.trigger_lambda()
+            except Exception as e:
+                logging.exception("Unable to trigger ingest lambda.\n%s" % e)
```

Comment:
catch an exception here so that other batches have a chance of getting kicked off.

we'll probably want to figure out how to get splunk to watch for these errors specifically, since that would prevent ingestion from ever happening.